### PR TITLE
Scale testing improvements

### PR DESCRIPTION
**Pull request by dgoodwin**
Various improvements for scale testing, updated instructions and observations. Details in individual commits.

### CODE REVIEW

File: contrib/pkg/createcluster/create.go [hunk 1]
```diff
@@ -260,6 +264,8 @@ create-cluster CLUSTER_DEPLOYMENT_NAME --cloud=ovirt --ovirt-api-vip 192.168.1.2
 	flags.StringVar(&opt.ManifestsDir, "manifests", "", "Directory containing manifests to add during installation")
 	flags.StringVar(&opt.MachineNetwork, "machine-network", "10.0.0.0/16", "Cluster's MachineNetwork to pass to the installer")
 	flags.StringVar(&opt.Region, "region", "", "Region to which to install the cluster. This is only relevant to AWS, Azure, and GCP.")
+	flags.StringSliceVarP(&opt.Labels, "labels", "l", []string{}, "Label to apply to the ClusterDeployment (key=val)")
```
Comment (staebler) [inline contrib/pkg/createcluster/create.go:267]:
```suggestion
	flags.StringSliceVarP(&opt.Labels, "labels", "l", nil, "Label to apply to the ClusterDeployment (key=val)")
```

File: contrib/pkg/createcluster/create.go [hunk 2]
```diff
@@ -383,6 +389,14 @@ func (o *Options) Validate(cmd *cobra.Command) error {
 		}
 	}
 
+	if len(o.Labels) > 0 {
```
Comment (staebler) [inline contrib/pkg/createcluster/create.go:392]: This check is not needed.

File: hack/scaletest/syncset-template.yaml [hunk 3]
```diff
@@ -15,15 +15,139 @@ objects:
...
+        a: asdddaasdjksdhlkajshdkljasdhlaksjdhlaksjdhlaksdhlaksjdhlaksjdhlkasjdhlkajsdhlaksjdhalksdjhalkdsjhlakshdhlkajssdhlkajsdhlaksjdhlkasjdhalksjdhalksjh
+        b: asdhakjsdhlaksdhlaksjsdhlkasjdhlkashdlkasdhlaksjdhlaksdhalskdjhalkdhaksljdhalksjdhaklsjdh
+        c: asjkdhaskjdhalksjdhaklsjdhlkjhllljkkkkkkkkkkkkkkkkkkkkkkhddhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh
+        d: asdiuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuu
+    resourceApplyMode: Sync
     clusterDeploymentSelector:
       matchLabels:
-        scaletest: "true"
+        # Added automatically by hiveutil create-cluster, so will match what we create in the adoption script:
+        hive.openshift.io/hiveutil-created: "true"
```
Comment (staebler) [inline hack/scaletest/syncset-template.yaml:153]: I don't understand the point of this change.

File: pkg/clusterresource/builder.go [hunk 4]
```diff
@@ -106,6 +106,9 @@ type Builder struct {
 
 	// MachineNetwork is the subnet to use for the cluster's machine network.
 	MachineNetwork string
+
+	// SkipMachinePools should be true if you do not want to manage MachineSets in-cluster once it is installed.
```
Comment (staebler) [inline pkg/clusterresource/builder.go:110]: The term "in-cluster" is confusing here.
```suggestion
	// SkipMachinePools should be true if you do not want to Hive to manage MachineSets once the cluster is installed.
```

File: pkg/clusterresource/builder.go [hunk 5]
```diff
@@ -157,8 +160,8 @@ func (o *Builder) Build() ([]runtime.Object, error) {
 	var allObjects []runtime.Object
 	allObjects = append(allObjects, o.generateClusterDeployment())
 
-	if mp := o.generateMachinePool(); mp != nil {
-		allObjects = append(allObjects, mp)
+	if !o.SkipMachinePools {
+		allObjects = append(allObjects, o.generateMachinePool())
```
Comment (staebler) [inline pkg/clusterresource/builder.go:164]: Please keep the nil check to be able to support platforms that do not support MachinePools.

File: pkg/clusterresource/builder.go [hunk 6]
```diff
@@ -341,8 +344,8 @@ func (o *Builder) generateInstallConfigSecret() (*corev1.Secret, error) {
 			Namespace: o.Namespace,
 		},
 		Type: corev1.SecretTypeOpaque,
-		Data: map[string][]byte{
-			"install-config.yaml": d,
+		StringData: map[string]string{
```
Comment (staebler) [inline pkg/clusterresource/builder.go:347]: What is the purpose of this change?

File: cmd/manager/main.go [hunk 7]
```diff
@@ -103,6 +103,13 @@ func newRootCommand() *cobra.Command {
 				log.WithError(err).Fatal("Cannot parse log level")
 			}
 			log.SetLevel(level)
+
+			// Add some millisecond precision to log timestamps, useful for debugging performance.
```
Comment (staebler) [inline cmd/manager/main.go:107]: Do we want the millisecond precision all the time, or just when we are debugging performance?

Comment (dgoodwin) [reply cmd/manager/main.go:107]: It's not critical but I felt it was probably useful and harmless to keep.

File: pkg/clusterresource/builder.go (hunk 6 above)
Comment (dgoodwin) [reply pkg/clusterresource/builder.go:347]: This allows us to create-cluster with -o yaml and the edit the install config without having to base64 decode the embedded secret data.

**Comment by dgoodwin** [comment]
Updated.

File: cmd/manager/main.go (hunk 7 above)
Comment (staebler) [reply cmd/manager/main.go:111]: Sounds good.

File: pkg/clusterresource/builder.go (hunk 6 above)
Comment (staebler) [reply pkg/clusterresource/builder.go:347]: Wonderful!
